ranger/container/directory.py

```diff
@@ -252,6 +252,19 @@ def hidden_filter_func(fobj):
                         return False
                 return True
             filters.append(hidden_filter_func)
+        if self.settings.global_inode_type_filter and not self.inode_type_filter:
+            # pylint: disable=too-many-boolean-expressions
+            def hidden_inode_filter_func(obj):
+                g_filter = self.settings.global_inode_type_filter
+                if (
+                        ('d' in g_filter and obj.is_directory) or
```

Comment:
The first version would be my choice as well.

As for the code, my understanding is that the filter does not 'exist' in ranger's code until the command is run, which is a problem for this use case.